### PR TITLE
Split out the recommendation tests

### PR DESCRIPTION
**Pull request by jaredpar**
The recommendation tests can be run in parallel but were in an assembly which
had many other tests that could not.  Splitting out into a separate assembly
which is designed for parallel runs.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-ide, @tannergooding

**Comment by jaredpar** [comment]
test win unit32 please

**Comment by jasonmalinowski** [comment]
How much work would it be to just fix this for real? Or, looking at xUnit's documentation, how much work would it be to just add all of our tests in CSharpEditorServicesTest into a single test collection so they don't run in parallel, and then we can remove the recommender tests from that collection to start getting the real parallelization benefit?

File: src/EditorFeatures/CSharpTest2/CSharpServicesTest.Debug.xunit [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<xunit>
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest2/CSharpServicesTest.Debug.xunit]: We don't use .xunit files for anything else, do we?

File: src/EditorFeatures/CSharpTest2/app.config [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  <appSettings>
+    <add key="xunit.useAppDomain" value="true"/>
+    <add key="xunit.diagnosticMessages" value="false"/>
+    <add key="xunit.parallelizeTestCollections" value="true"/>
+  </appSettings>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="System.Collections.Immutable" publicKeyToken="b03f5f7f11d50a3a" culture="neutral"/>
+        <bindingRedirect oldVersion="1.0.0.0-1.65535.65535.65535" newVersion="1.1.37.0"/>
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest2/app.config]: I don't think we want this, do we?

Comment (jaredpar) [inline src/EditorFeatures/CSharpTest2/app.config]: I just copied the same file from CSharpTest.  So if this is wrong then htey're both wrong.

File: src/EditorFeatures/CSharpTest2/CSharpServicesTest.Debug.xunit (hunk 1 above)
Comment (jaredpar) [inline src/EditorFeatures/CSharpTest2/CSharpServicesTest.Debug.xunit]: I don't think there are used anymore.  We can probably just delete them in mass from the repo.

**Pull request closed without merging**